# Chi-Square Test for Independence
# Mizzare Corporation - Device Type vs. Customer Satisfaction

### Hypotheses

**Background:**
Mizzare Corporation wants to check if there's a significant association between device type (Smart Thermostats vs. Smart Lights) and customer satisfaction levels.

**Hypotheses:**
- H₀ (Null): Device type and customer satisfaction are independent.
- H₁ (Alternative): Device type and customer satisfaction are associated (not independent).


### Input data

In [5]:
import numpy as np
import pandas as pd

# Observed data
data = {
    "Satisfaction": ["Very Satisfied", "Satisfied", "Neutral", "Unsatisfied", "Very Unsatisfied"],
    "Smart Thermostat": [50, 80, 60, 30, 20],
    "Smart Light": [70, 100, 90, 50, 50]
}

df = pd.DataFrame(data)
df["Total"] = df["Smart Thermostat"] + df["Smart Light"]
df.loc["Total"] = df[["Smart Thermostat", "Smart Light", "Total"]].sum()
df

,Satisfaction,Smart Thermostat,Smart Light,Total
0,Very Satisfied,50.0,70.0,120.0
1,Satisfied,80.0,100.0,180.0
2,Neutral,60.0,90.0,150.0
3,Unsatisfied,30.0,50.0,80.0
4,Very Unsatisfied,20.0,50.0,70.0
Total,NaN,240.0,360.0,600.0


###  Expected Counts & Chi-Square Statistic

In [6]:
# Extract observed counts
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
], dtype=float)

# Totals
row_totals = observed.sum(axis=1)
col_totals = observed.sum(axis=0)
grand_total = observed.sum()

# Expected counts
expected = np.outer(row_totals, col_totals) / grand_total

# Chi-Square statistic
chi_square_stat = ((observed - expected) ** 2 / expected).sum()

pd.DataFrame(expected,
             columns=["Smart Thermostat", "Smart Light"],
             index=["Very Satisfied", "Satisfied", "Neutral", "Unsatisfied", "Very Unsatisfied"])

,Smart Thermostat,Smart Light
Very Satisfied,48.0,72.0
Satisfied,72.0,108.0
Neutral,60.0,90.0
Unsatisfied,32.0,48.0
Very Unsatisfied,28.0,42.0


### Critical Value & P-Value

In [7]:
from scipy.stats import chi2

# Degrees of freedom = (r-1)*(c-1)
r, c = observed.shape
df_chi = (r - 1) * (c - 1)

alpha = 0.05
critical_value = chi2.ppf(1 - alpha, df_chi)
p_value = 1 - chi2.cdf(chi_square_stat, df_chi)

chi_square_stat, critical_value, p_value

(np.float64(5.638227513227513),
 np.float64(9.487729036781154),
 np.float64(0.22784371130697179))

### Decision & Conclusion

In [8]:
if chi_square_stat > critical_value:
    decision = "Reject H0"
    conclusion = "There is a significant association between device type and customer satisfaction."
else:
    decision = "Fail to Reject H0"
    conclusion = "There is no significant association between device type and customer satisfaction."

decision, conclusion

('Fail to Reject H0',
 'There is no significant association between device type and customer satisfaction.')

### Final Output Summary

- Chi-Square Statistic: ~5.64  
- Degrees of Freedom: 4  
- Critical Value (α = 0.05): ~9.49  
- P-value: ~0.228  

**Decision:** Fail to Reject H₀  
**Conclusion:** There is not enough evidence to conclude that device type and customer satisfaction are associated.
